provider/common/helpers.go

```diff
@@ -194,3 +199,9 @@ func Sha1Hash(data []byte) (string, error) {
 func HasValue(val attr.Value) bool {
 	return !val.IsUnknown() && !val.IsNull()
 }
+
+func AssertStateAndPlanValue(stateAttr attr.Value, planAttr attr.Value, attrName string, diags *diag.Diagnostics) {
```

Comment:
The Assert prefix says that this function should return an error in case the condition is false
Why don't you return error?
Consider  the name: `ValidateEquals`
And add description: `ValidateEquals checks if given attributes are equal, if not add error to diagnostic`

So also that should affect caller function, change it to validate instade of assert

Reply:
I changed to `ValidateStateAndPlanEquals`